### PR TITLE
Speed up writeGroupVInts

### PR DESCRIPTION
**Pull request by easyice**
This change uses `VarHandles` instead of `BytesRefBuilder#append` to speed up `writeGroupVInts`, which is more aligned with group-varint generally implementation.

Main:
```
Benchmark                                (size)   Mode  Cnt  Score   Error   Units
GroupVIntBenchmark.bench_writeGroupVInt      64  thrpt    5  1.050 ± 0.057  ops/us
```

PR:
```
Benchmark                                (size)   Mode  Cnt  Score   Error   Units
GroupVIntBenchmark.bench_writeGroupVInt      64  thrpt    5  2.924 ± 0.049  ops/us
```

### CODE REVIEW

File: lucene/core/src/java/org/apache/lucene/util/GroupVIntUtil.java [hunk 1]
```diff
@@ -111,4 +112,40 @@ public static int readGroupVInt(
     pos += 1 + n4Minus1;
     return (int) (pos - posStart);
   }
+
+  private static int numBytes(int v) {
+    // | 1 to return 1 when v = 0
+    return Integer.BYTES - (Integer.numberOfLeadingZeros(v | 1) >> 3);
+  }
+
+  public static void writeGroupVInts(DataOutput out, byte[] scratch, long[] values, int limit)
```
Comment (jpountz) [inline lucene/core/src/java/org/apache/lucene/util/GroupVIntUtil.java:121]: Can you add some minimal javadocs?

**Review by jpountz — APPROVED** [review]
This looks like a good speedup!

File: lucene/core/src/java/org/apache/lucene/util/GroupVIntUtil.java (hunk 1 above)
Comment (easyice) [reply lucene/core/src/java/org/apache/lucene/util/GroupVIntUtil.java:125]: done!

**Comment by easyice** [comment]
Thanks for reviewing!

**Review by jpountz — APPROVED** [review]

**Pull request merged**